src/modules/components/member/memberAvatar/memberAvatar.tsx

```diff
@@ -0,0 +1,63 @@
+import * as blockies from 'blockies-ts';
+import classNames from 'classnames';
+import type React from 'react';
+import { isAddress, type Hash } from 'viem';
+import { useEnsAddress, useEnsAvatar, useEnsName } from 'wagmi';
+import { Avatar, Spinner, type IAvatarProps } from '../../../../core';
+
+export interface IMemberAvatarProps extends Omit<IAvatarProps, 'fallback'> {
+    /**
+     * ENS name of the user to lookup avatar src.
+     */
+    ensName?: string;
+    /**
+     * 0x address of the user to look up ENS name and avatar src.
+     */
+    address?: string;
+    /**
+     * Direct URL src of the user avatar image to be rendered.
+     */
+    avatarSrc?: string;
+}
+
+export const MemberAvatar: React.FC<IMemberAvatarProps> = (props) => {
+    const { ensName, address, avatarSrc, size = 'sm', className, ...otherProps } = props;
+    const isValidAddress = address ? isAddress(address) : false;
+    const isValidENSName = ensName ? ensName.length >= 7 && ensName.endsWith('.eth') : false;
```

Comment:
Can be simplified as:
```suggestion
    const isValidAddress = address != null && isAddress(address);
    const isValidENSName = ensName != null && ensName.length >= 7 && ensName.endsWith('.eth');
```